hedera-node/src/main/java/com/hedera/services/store/contracts/ContractsStore.java

```diff
@@ -147,17 +149,27 @@ public void clearStorage(Address address) {
 
 	@Override
 	public void commit() {
-		provisionalAccountUpdates.forEach((address, evmAccount) -> {
-			final var accId = EntityIdUtils.accountParsedFromSolidityAddress(address.toArray());
-			if (ledger.exists(accId)) {
-				final var account = ledger.get(accId);
-				ledger.adjustBalance(accId, evmAccount.getBalance().toLong() - account.getBalance());
-			} else {
-				var pair = this.provisionalAccountCreations.get(accId);
-				ledger.create(pair.getKey(), accId, evmAccount.getBalance().toLong(), pair.getValue());
+		provisionalAccountCreations.forEach((accountId, entry) -> {
+			if (!ledger.exists(accountId)) {
+				var pair = entry;
+				if (pair != null) {
+					if (pair.getValue().getChanges().get(AccountProperty.IS_SMART_CONTRACT).equals(true)) {
+						/* Allow creation only for contract accounts */
+						ledger.create(pair.getKey(), accountId, 0L, pair.getValue());
+					} else {
+						// TODO throw exception, account creation not allowed, resolve after refactoring
+					}
+				}
 			}
 		});
 
+		provisionalAccountUpdates.forEach((address, evmAccount) -> {
+			final var accountId = EntityIdUtils.accountParsedFromSolidityAddress(address.toArray());
+			if (ledger.exists(accountId)) {
+				final var account = ledger.get(accountId);
+				ledger.adjustBalance(accountId, evmAccount.getBalance().toLong() - account.getBalance());
+			}
```

Comment:
we can add TODO in the `else` -> `//TODO throw exception` after @rusiboyadjiev 's PR

Reply:
- the simple transfer test does a `call` 
- there is a TODO in the creations logic, do we need it also in the updates?